### PR TITLE
Fix requester pays check#824

### LINKED ISSUE #824: As of 2022-02-22 Automatic detection of requesterPays buckets is failing
First observed on PR https://github.com/googleapis/java-storage-nio/pull/823 and visible in the integration tests related to Requester Pays ([build results](https://source.cloud.google.com/results/invocations/aaeb196d-16d2-45c9-8104-48dbc5a8c855/targets/github%2Fjava-storage-nio%2Fgoogle-cloud-nio%2Ftarget%2Ffailsafe-reports%2Fintegration%2Fcom.google.cloud.storage.contrib.nio.it.ITGcsNio/tests))

#### Environment details

1. this library
2. OS type and version: N/A
3. Java version: N/A
4. version(s): all released prior to 2022-02-22

#### Code example

https://github.com/googleapis/java-storage-nio/blob/417704a68c9fe8ad618300d23209364066bd9826/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java#L310-L316

#### External references such as API reference guides

It seems the error message response changed subtly
```diff
- Bucket is requester pays bucket but no user project provided.
+ Bucket is a requester pays bucket but no user project provided.
```

According to [public documentation](https://cloud.google.com/storage/docs/json_api/v1/status-codes#400_Bad_Request), we should expect the `reason` to be `userProjectMissing`, however it is `required` as can be seen in 
```json
{
  "error": {
    "code": 400,
    "message": "Bucket is a requester pays bucket but no user project provided.",
    "errors": [
      {
        "message": "Bucket is a requester pays bucket but no user project provided.",
        "domain": "global",
        "reason": "required"
      }
    ]
  }
}
```

Related from NodeJS https://github.com/googleapis/nodejs-storage/pull/1791/files

### PR DESCRIPTION
**Pull request by lbergelson**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage-nio/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass <- Normal tests and linter pass but I was unable to configure a service account correctly to pass the integration tests.
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #824 ☕️

* The error message received when querying for requester pays status changed to include an 'a' which caused strict matching to fail. 
* This fixes and standardizes the error message to use the same string for both the code and the tests.

### CODE REVIEW

**Comment by lbergelson** [comment]
If someone knows where that error message is generated so we could refer to it directly it would be good....

**Comment by lbergelson** [comment]
@droazen That seems reasonable assuming they pass now.

File: google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java [hunk 1]
```diff
@@ -87,6 +87,8 @@
 @ThreadSafe
 public final class CloudStorageFileSystemProvider extends FileSystemProvider {
 
+  public static final String BUCKET_IS_REQUESTER_PAYS_ERROR_MESSAGE =
+      "Bucket is a requester pays bucket but no user project provided";
```
Comment (kshakir) [inline google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java:91]: @lbergelson have you considered a simpler match string that may have a better chance of not breaking in the future?

For example, it looks like @breilly2 and @mcovarr updated cromwell to use "requester pays bucket but no user project": 
https://github.com/broadinstitute/cromwell/pull/6689

Meanwhile terra-ui searches for the even more basic string "requester pays", and didn't even need to be updated with the last server-side change: https://github.com/DataBiosphere/terra-ui/blob/e34bdf0b81d7d2750787108ac06238787a363726/src/libs/ajax.js#L76

**Comment by sydney-munro** [comment]
We are working with the API team to provide a more permanent solution to this issue and with a fallback to checking the error message. 

See: https://github.com/googleapis/java-storage-nio/pull/841

**Comment by lbergelson** [comment]
@kshakir Yeah, I was going to reduce it to just "requester pays" but I figured the smaller the change the faster it might get merged.  Those extra few characters are hard to review!

**Comment by lbergelson** [comment]
@sydney-munro Thank you, that's great.  It's a very fragile mechanism at the moment. Even fixing it so that the storage service returns a better structured error message is still a bit awkward though.  The biggest remaining issue is that the call requires `storage.bucket.sget` permissions to work at all which is not always available ( and isn't necessary to read the file) just check requester pays status.  It seems like it should be altered to not have to require that permission.  Getting this fixed quickly though is the priority.

**Comment by droazen** [comment]
@sydney-munro Sounds good, but what's the timeline for merging the more permanent solution? If it's more than a week or so, could we possibly merge the patched error message check as a stopgap measure just to get this feature working again? Thanks!

**Comment by droazen** [comment]
@lbergelson Since https://github.com/googleapis/java-storage-nio/pull/841 was just merged, I think this can be closed.

**Pull request closed without merging**